ui/panels-plugin/src/plugins/gauge-chart/thresholds.ts

```diff
@@ -11,20 +11,73 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-export const ThresholdColorsMap = {
-  green: 'rgb(115, 191, 105)',
-  orange: 'rgba(237, 129, 40, 0.89)',
-  red: 'rgba(245, 54, 54, 0.9)',
+import { merge } from 'lodash-es';
+
+export const ThresholdColors = {
+  GREEN: 'rgba(115, 191, 105, 1)',
+  ORANGE: 'rgba(237, 129, 40, 0.9)',
+  RED: 'rgba(245, 54, 54, 0.9)',
 };
 
-export type ThresholdColorsType = keyof typeof ThresholdColorsMap;
+export type ThresholdColorsType = keyof typeof ThresholdColors | string;
 
 export type ThresholdOptions = {
   steps: StepOptions[];
+  default_color?: string;
 };
 
-type StepOptions = {
-  // color: string;
-  color: ThresholdColorsType;
+export type StepOptions = {
   value: number;
+  color: ThresholdColorsType;
 };
+
+export function convertThresholds(
+  thresholds: ThresholdOptions = {
+    steps: [{ value: 0, color: ThresholdColors.GREEN }],
+  }
+) {
+  const defaultThresholdColor = thresholds.default_color || ThresholdColors.GREEN;
+  const defaultThresholdArr: [number, string] = [0, defaultThresholdColor];
+  const defaultWarningColor = ThresholdColors.ORANGE;
+  const defaultAlertColor = ThresholdColors.RED;
+
+  const defaultThresholds: ThresholdOptions = {
```

Comment:
Yeah I agree, I went ahead and made thresholds optional--it ended up being easier to do this with the zip approach instead of merging defaultThresholds like it was before.